### ISSUE TITLE
What is NN structure?

### ISSUE THREAD

**Issue by JBMing** [tambetm/matlab2048#1]
I did't find the information of neural network such as the number of layers and the neurons in each layer.Thus,can u explain the explicit information in readme?

**Comment by tambetm** [comment]
Currently the NN has only fully connected layers. The sizes of hidden layers are passed to `NNAgent` constructor as `opts.layers` parameter. For example `opts.layers = [1000];` creates one hidden layer with 1000 nodes, `opts.layers = [256 256];` creates two hidden layers with 256 nodes each. README seems outdated, but example.m should be OK. Will fix it ASAP.

**Comment by JBMing** [comment]
Do you mean just one fully connected layers and not have convolutional layer?
Besides, I am confused by the readme and don't how execute the code.
Whether should I install some other environment?

**Comment by tambetm** [comment]
You can have several fully connected layers by including many numbers in list, i.e. `opts.layers = [256 256]`. But no convolutional layers. 

I would suggest to check out example code in https://github.com/tambetm/matlab2048/blob/master/example.m, that should work. 

You need to check out DeepLearnToolbox from GitHub as suggested in README. No other packages are needed.

**Comment by JBMing** [comment]
After i download and use addpath to install the DeepLearnToolbox, the example.m code still can't work.
The error happening in nnff.mat and it is come from beyond the matrix index.
Could u explain it?

**Comment by JBMing** [comment]
I am trying to modify the algorithm of weight updation.
But I have question for the code. 
Where is the code that achieve back=propagation and gradient descent ?
Could u tell me ?

**Comment by tambetm** [comment]
Backpropagation and gradient descent are implemented in [DeepLearnToolbox](https://github.com/rasmusbergpalm/DeepLearnToolbox). I'm using [my own fork](https://github.com/tambetm/DeepLearnToolbox), which has some minor modifications, not relevant to this project AFAIK. The backpropagation is implemented in [NN/nnff.m](https://github.com/tambetm/DeepLearnToolbox/blob/master/NN/nnff.m), [NN/nnbp.m](https://github.com/tambetm/DeepLearnToolbox/blob/master/NN/nnbp.m) and [NN/nnapplygrads.m](https://github.com/tambetm/DeepLearnToolbox/blob/master/NN/nnapplygrads.m).
